docs/DeveloperGuide.md

```diff
@@ -1,38 +1,416 @@
+* `Manager`
+    * On creation: Initializes the Profile, TextUI and Storage components.
+    * On run: Loads data from `Storage`, receives user input from UI and uses `Parser` to parse user input into application.
+* `Profile`
+    * Handles cached data within run time of application. Manages and stores the list of `session` 
+      that were created by the user before storing it into `Storage`.
+* `TextUI`
+    * Handles UI operations of the application.
+* `Storage`
+    * Handles storage operations of the application.
+* `Parser`
+    * Handles input parsing and determines which command to run.
+* `Command`
+    * Defines how a command is executed.
+
+**Interaction between components**
+![Component Interaction Screenshot](https://raw.githubusercontent.com/AY2122s2-cs2113t-t10-1/tp/master/docs/images/developerguide/ComponentInteraction.drawio.png)
+<br>
+The *Component Interaction Diagram* shows the inner workings of how each component in SplitLah interacts.
+The diagram depicts a scenario when a user attempts to create a session.
+
+
+
+### SplitLah Component
+![SplitLah Component Screenshot](https://raw.githubusercontent.com/AY2122s2-cs2113t-t10-1/tp/master/docs/images/developerguide/SplitLahComponent.drawio.png)
+<br>
+The `SplitLah` component is the application's main class. Its job is to initialize an instance of `Manager` when the
+application starts. After initialization, it would then proceed to run a loop which would prompt the user for a 
+command. When it receives a command from the user, it would invoke the `parser` and retrieve the command for SplitLah
+to run. Upon using the `Exit` command, SplitLah would then exit from the command loop and end the application.
+
+
+### Manager Component
+![Manager Component Screenshot](https://raw.githubusercontent.com/AY2122s2-cs2113t-t10-1/tp/master/docs/images/developerguide/ManagerComponent.drawio.png)
+<br>
+The `Manager` class is initialized by the `SplitLah` class (the main class) when the application starts.
+It stores the `Profile`, `TextUI` and `Storage` objects. The `Profile` class helps to manage all data accesses 
+throughout the lifetime of the application. While the `Storage` helps to save what the `Profile` class has captured. 
+The `TextUI` class serves as an interface to read user inputs and print application outputs.
+
+### Profile Component
+![Profile Component Screenshot](https://raw.githubusercontent.com/AY2122s2-cs2113t-t10-1/tp/master/docs/images/developerguide/Profile%20Component.drawio.png)
+<br>
+The `Profile` class holds the list of sessions and groups that have been created by the user. 
+It also tracks the unique identifier for `Session`, `Activity` and `Group` classes. The `Profile`
+class would return a unique identifier every time a new `Session`, `Activity` or `Group` is created. 
+
+### TextUI Component
+![TextUI Component Screenshot](https://raw.githubusercontent.com/AY2122s2-cs2113t-t10-1/tp/master/docs/images/developerguide/TextUI%20Component.drawio.png)
```

Comment:
Screenshot for TextUI component does not load:
![image](https://user-images.githubusercontent.com/69501969/160956660-cc01160b-4c73-4fc2-8042-dd038d21ab88.png)